SQLite/Core/Connection.swift

```diff
@@ -25,8 +25,209 @@
 import Dispatch
 import CSQLite
 
+
+/// The mode in which a transaction acquires a lock.
+public enum TransactionMode : String {
+    
+    /// Defers locking the database till the first read/write executes.
+    case Deferred = "DEFERRED"
+    
+    /// Immediately acquires a reserved lock on the database.
+    case Immediate = "IMMEDIATE"
+    
+    /// Immediately acquires an exclusive lock on all databases.
+    case Exclusive = "EXCLUSIVE"
+    
+}
+
+
+/// Protocol to an SQLite connection
+public protocol Connection {
```

Comment:
Yep, this is the proposal: https://github.com/apple/swift-evolution/blob/master/proposals/0006-apply-api-guidelines-to-the-standard-library.md

> Strip `Type` suffix from protocol names. In a few special cases this means adding a `Protocol` suffix to get out of the way of type names that are primary (though most of these we expect to be obsoleted by Swift 3 language features).

I need to go back and update the other `Type`-suffixed protocols, as well.